graphs/possible_bipartition.py

```diff
@@ -5,8 +5,53 @@ def possible_bipartition(dislikes):
     """ Will return True or False if the given graph
         can be bipartitioned without neighboring nodes put
         into the same partition.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(E * N)
+        Space Complexity: O(n)
     """
-    pass
+    if not dislikes:
+        return True
+    
+    to_pet = deque() # O(n)
+    pet_dogs = set() # O(n)
+    playpen_1 = set() # O(n)
+    playpen_2 = set() # O(1) if above is O(n)
 
+    # Find the first dog with dog rivals ... O(E N)
+    for dog in range(len(dislikes)):
+        if to_pet:
+            break
+        if not dislikes[dog]:
+            pet_dogs.add(dog)
+        if dog:
+            for each_disliked_dog in dislikes[dog]:
+                to_pet.append(each_disliked_dog)
```

Comment:
Here you append each of the rivals of the first dog with rivals to the queue. From your comment on line 19, it seems like you may have intended to just append `dog` aka the first dog with rivals instead.

Additionally, `if dog` will evaluate to `True` except for dog `0` since `0` is a Falsey value in Python. I think what you intended to check is whether the dog had rivals which would be `if dislikes[dog]`

This doesn't really affect your code below,  but it may slightly mess up the order of BFS since you may have multiple elements in your `to_pet` queue to start.

```suggestion
        else:
            to_pet.append(dog)
```